blob_writer.js

```diff
@@ -62,11 +108,12 @@ function write_file_via_bridge({
         recursive,
         data: ""
     }).then(function ({uri}) {
+        return update_blob_content(directory, path, blob).catch(async () => {
```

Comment:
We only truncate the file so that we can run `append_blob` multiple times. For the web platform, we can just overwrite the blob directly. I would leave `write_file_via_bridge` as is, and create a new function `write_file_via_indexeddb` to handle the new functionality. Use `Capacitor.platform` to determine the strategy, rather than depending on exceptions.

Reply:
We should use `Capacitor.getPlatform()` instead of `Capacitor.platform` since the later is deprecated.